MimeKit/Cryptography/BouncyCastleSecureMimeContext.cs

```diff
@@ -1300,7 +1233,27 @@ Stream Envelope (CmsRecipientCollection recipients, Stream content, Cancellation
 
 			foreach (var recipient in recipients) {
 				if (unique.Add (recipient.Certificate)) {
-					cms.AddRecipientInfoGenerator (new CmsRecipientInfoGenerator (recipient));
+					var cert = recipient.Certificate;
+					var pub = recipient.Certificate.GetPublicKey();
+					if (pub is RsaKeyParameters) {
+						cms.AddKeyTransRecipient (cert);
```

Comment:
I think the reason I had the CmsRecipientInfoGenerator originally (and I'll have to verify by reviewing commit logs) was because I needed to be able to:

1. allow users to specify SubjectKeyIdentifier vs IssuerAndSerialNumber (the 2 possible values of the CmsRecipient.RecipientIdentifierType property)
2. allow users to override the CmsRecipient.RsaEncryptionPadding

CmsEnvelopeInfoGenerator.AddKeyTransRecipient() didn't appear to allow that from what I could tell.

I believe that's why I had to do my own thing.

Reply:
The first, a new BC API call to specify the `RecipientInfo` might be nice in that case.

Reply:
@jstedfast Aha! So I took a second, closer look at the APIs. I think we can already do this without needing the hook! There's two APIs:

```c#
public void AddKeyTransRecipient(X509Certificate cert) -> KeyTransRecipientInfoGenerator -> IssuerAndSerialNumber(recipCert.CertificateStructure)
```

vs

```c#
public void AddKeyTransRecipient(AsymmetricKeyParameter pubKey, byte[] subKeyId) -> KeyTransRecipientInfoGenerator -> byte[] subjectKeyID, IKeyWrapper keyWrapper
```

I think these do what you want with RecipientInfo construction. The same applies to the two forms of `AddKeyAgreementRecipient` as well :-) 


So back to looking for a viable OAEP approach.